### PR TITLE
672 change tokens consumption

### LINKED ISSUE #672: Change the way the website consumes/displays tokens per version
**Is your feature request related to a problem? Please describe.**
Since we changed to tokens 2.0, we need to update the website to handle the new structure and files that are POSTed with the documentation from each code repo.

**Describe the solution you'd like**
Update the code to handle the new structure (and maybe the old too? maybe not). Also note we do not use `.raw.json`, its just `.json` now.

**Describe alternatives you've considered**
n/a

**Additional context**

Pre-req:
- https://github.com/infor-design/css/issues/176
- https://github.com/infor-design/enterprise/issues/951

### PR DESCRIPTION
**Pull request by jmacaluso711**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
updates the token service to take the new token.simple.json structure and minor modifications to the token table

**Related github/jira issue (required)**:
closes #672 

**Steps necessary to review your pull request (required)**:
* pull branch
* publish docs from `ids-identity@2.0.0-alpha.1`
* check that the token table is rendering properly in the css docs http://localhost/code/ids-css/latest/button
* For design core content pages login to CMS and change `tokensCategory` to singular and check http://localhost/guidelines/components/elements/buttons

### CODE REVIEW

File: src/web/src/app/components/docs-content-page/docs-content-page.component.ts [hunk 1]
```diff
@@ -45,6 +45,7 @@ export class DocsContentPageComponent implements OnInit, OnDestroy {
   public tocItems: TocItems[] = [];
   public bodyTitles;
   public apiTitles;
+  public tokensCategory;
```
Comment (clepore) [inline src/web/src/app/components/docs-content-page/docs-content-page.component.ts]: I don't think we need this for anything?

File: src/web/src/app/shared/token.service.spec.ts [hunk 2]
```diff
@@ -80,24 +80,45 @@ describe('TokenService', () => {
...
-        'token-three': {
-          'category': 'categoryA',
-          'name': 'token-three'
+        'value': 'token value',
+        'type': 'token type',
+        'orignal': {
+          'value': 'original value'
         }
-      }
+      },
```
Comment (clepore) [inline src/web/src/app/shared/token.service.spec.ts]: Remove extra comma

File: src/web/src/app/shared/token.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
-  category: string;
-  description: string;
   name: string;
-  originalValue: string;
-  type: string;
   value: string;
+  category: string;
+  original: {
+    value: string
+  };
```
Comment (clepore) [inline src/web/src/app/shared/token.ts]: Do we need to update `name` here as an object with 3 properties under it?
```
{
  name: {
    human: string
    sass: string,
    javascript: string
  };
  value: string;
  category: string;
  type: string;
  original: {
    value: string
  };
}
```

File: src/web/src/app/shared/token.service.ts [hunk 4]
```diff
@@ -50,17 +52,15 @@ export class TokenService {
...
-    return str.replace(/\w\S*/g, (txt) => {
-      return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
+  /**
+   * For CMS pages filter by the tokensCategory dot notation string. (theme, theme.font, etc.)
+   * @param tokenData response from ${domain}/api/docs/${library}/${version}/tokens/web/theme-soho.simple.json
+   * @param cmsInput input from tokensCategory field in the CMS.
+   */
+  filterCmsTokens(tokenData, cmsInput) {
+    const tokens = tokenData.filter(token => {
+      return token.name.javascript.includes(cmsInput);
```
Comment (nickwynja) [inline src/web/src/app/shared/token.service.ts]:
```suggestion
      return token.name.javascript.startsWith(cmsInput);
```

File: src/web/src/app/components/core-content-page/core-content-page.component.ts [hunk 5]
```diff
@@ -36,6 +36,7 @@ export class CoreContentPageComponent implements AfterViewInit {
           this.pageContent = res;
         },
         err => {
+          console.error(err);
```
Comment (nickwynja) [inline src/web/src/app/components/core-content-page/core-content-page.component.ts]:
```suggestion
```

File: src/web/src/app/components/element-page/element-page.component.ts [hunk 6]
```diff
@@ -36,6 +36,7 @@ export class ElementPageComponent implements AfterViewInit {
             this.pageContent = res;
           },
           err => {
+            console.error(err);
```
Comment (nickwynja) [inline src/web/src/app/components/element-page/element-page.component.ts]:
```suggestion
```

File: src/web/src/app/components/block-page/block-page.component.ts [hunk 7]
```diff
@@ -36,6 +36,7 @@ export class BlockPageComponent implements AfterViewInit {
             this.pageContent = res;
           },
           err => {
+            console.error(err);
```
Comment (nickwynja) [inline src/web/src/app/components/block-page/block-page.component.ts]:
```suggestion
```

File: src/web/src/app/components/token-table/token-table.component.ts [hunk 8]
```diff
@@ -25,15 +24,22 @@ export class TokenTableComponent implements OnInit {
...
+          res => {
+            if (this.cms) {
+              this.idsTokenProperties = this.tokenService.filterCmsTokens(res, this.tokensCategory);
+            } else {
+              this.idsTokenProperties = this.tokenService.groupTokensByCategory(res);
+            }
+            this.loading = false;
+          },
+          err => {
+            console.log(`No tokens found: ${err}`);
```
Comment (nickwynja) [inline src/web/src/app/components/token-table/token-table.component.ts]:
```suggestion
```

File: src/web/src/app/components/token-table/token-table.component.ts [hunk 9]
```diff
@@ -25,15 +24,22 @@ export class TokenTableComponent implements OnInit {
...
-        }
-      );
+      this.tokenService
+        .getTokenData(domain, library, version)
+        .subscribe(
+          res => {
+            if (this.cms) {
+              this.idsTokenProperties = this.tokenService.filterCmsTokens(res, this.tokensCategory);
+            } else {
+              this.idsTokenProperties = this.tokenService.groupTokensByCategory(res);
```
Comment (nickwynja) [inline src/web/src/app/components/token-table/token-table.component.ts]: I don't think we need `groupTokensByCategory()` anymore or need to know `this.cms` to switch off. The expectation is that we'd have something like `theme.color.font` in both the CMS and the code docs, so the front end can treat them the same way.

**Pull request merged**